pkg/util/steps/condition.go

```diff
@@ -66,9 +69,50 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
+		return c.enrichConditionTimeoutError()
+	}
 	return err
 }
 
 func (c conditionStep) String() string {
 	return fmt.Sprintf("[Condition %s, timeout %s]", friendlyName(c.f), c.timeout)
 }
+
+func (c conditionStep) enrichConditionTimeoutError() error {
+	funcNameParts := strings.Split(friendlyName(c.f), ".")
+	funcName := strings.Replace(funcNameParts[len(funcNameParts)-1], "-fm", "", 1)
+
+	message := "timed out polling for a prerequisite cluster condition to resolve"
+
+	// All functions passed as a `conditionFunction`: https://github.com/Azure/ARO-RP/blob/master/pkg/cluster/condition.go
+	// should have a case block below with a generic error message for the condition when it doesn't resolve
+
+	// TODO: Enrich the case-specific error messages below with more informative messages, as outlined in:
+	//  https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9274010
+	//  https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9405067
+	switch funcName {
+	case "apiServersReady":
+		message = "Kube API has not initialised successfully and is unavailable"
+	case "aroDeploymentReady":
+		message = "ARO Cluster Operator has failed to initialise successfully"
+	case "bootstrapConfigMapReady":
+		message = "bootstrap configuration required to create new cluster nodes is unavailable"
+	case "clusterVersionReady":
+		message = "Cluster Version Operator has not started successfully"
+	case "ingressControllerReady":
+		message = "Ingress Controller has not initialised successfully"
```

Comment:
https://askubuntu.com/questions/195073/synaptic-initialize-initialise-why-the-difference :thinking: 

`initialise` if region is located in Australia, otherwise `initialize` :joy: